pkg/arvo/app/http-server-store.hoon

```diff
@@ -0,0 +1,220 @@
+          [/ /'~landscape' /'~chat' /'~link' /'~dojo' /'~groups' /'~publish' ~]
+        |=(pax=path [pax [/app/landscape %.n]])
+      ==
+  :~  (connect /)
+      (connect /'~landscape')
+      (connect /'~chat')
+      (connect /'~link')
+      (connect /'~dojo')
+      (connect /'~groups')
+      (connect /'~publish')
+  ==
+  ::
+  ++  connect
+    |=  =path
+    ^-  card
+    [%pass path %arvo %e %connect [~ path] %http-server-store]
+  --
+::
+++  on-save  !>(state)
+++  on-load
+  |=  old-vase=vase
+  ^-  (quip card _this)
+  [~ this(state !<(state-zero old-vase))]
+::
+++  on-poke
+  |=  [=mark =vase]
+  ^-  (quip card _this)
+  |^
+  ?>  (team:title our.bowl src.bowl)
+  ?+  mark  (on-poke:def mark vase)
+      %http-server-action  (landscape-action !<(action vase))
+      %handle-http-request
+    =+  !<([id=@ta req=inbound-request:eyre] vase)
+    :_  this
+    %+  give-simple-payload:app  id
+    (handle-http-request req)
+  ==
+  ::
+  ++  landscape-action
+    |=  act=action
+    ^-  (quip card _this)
+    |^
+    ?-  -.act
+        %serve-dir
+      =*  url-base  url-base.act
+      ?:  (~(has by serving) url-base)
+        ~|("url already bound to {<(~(got by serving) url-base.act)>}" !!)
+      :-  [%pass url-base %arvo %e %connect [~ url-base] %http-server-store]~
+      this(serving (~(put by serving) url-base [clay-base.act public.act]))
+    ::
+        %unserve-dir
+      :-  [%pass url-base.act %arvo %e %disconnect [~ url-base.act]]~
+      this(serving (~(del by serving) url-base.act))
+    ::
+        %toggle-permission
+      ?.  (~(has by serving) url-base.act)
+        ~|("url is not bound" !!)
+      =/  [clay-base=path public=?]  (~(got by serving) url-base.act)
+      :-  ~
+      this(serving (~(put by serving) url-base.act [clay-base !public]))
+    ::
+        %set-landscape-homepage-prefix
+      =.  landscape-homepage-prefix.configuration  prefix.act
+      :_  this
+      (give [%configuration configuration])
+    ==
+    ::
+    ++  give
+      |=  =update
+      ^-  (list card)
+      [%give %fact [/all]~ [%http-server-update !>(update)]]~
+    --
+  ::
+  ++  handle-http-request
+    |=  =inbound-request:eyre
+    ^-  simple-payload:http
+    |^
+    =*  req       request.inbound-request
+    =*  headers   header-list.req
+    =/  req-line  (parse-request-line url.req)
+    ?.  =(method.req %'GET')  not-found:gen
+    =?  req-line  ?=(~ ext.req-line)
+      [[[~ %html] ~['index']] args.req-line]
+    ?>  ?=(^ ext.req-line)
+    ?~  site.req-line
+      not-found:gen
+    =*  url-prefix  landscape-homepage-prefix.configuration
+    ?.  ?|  ?&(?=(^ url-prefix) =((need url-prefix) i.site.req-line))
+            =(url-prefix ~)
+        ==
+      not-found:gen
+    ::
+    ?:  ?=([%'~landscape' %js %session ~] site.req-line)
+      %+  require-authorization-simple:app
+        inbound-request
+      %-  js-response:gen
+      (as-octt:mimes:html "window.ship = '{+:(scow %p our.bowl)}';")
```

Comment:
This is my fault, but I don't think we actually need this if we have the +who generator.